File: src/main/java/src/baseDatos/Serializador.java
package baseDatos;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;
import java.io.PrintWriter;
import java.util.ArrayList;

import gestorAplicacion.personas.*;
import gestorAplicacion.Calendario.*;


public class Serializador {

	    public static void serializar(gestionDatos gestor) {
	    	serializarEstudiantes(gestor,new File("src\\baseDatos\\temp\\Estudiantes.txt"));
	    	serializarProfesores(gestor,new File("src\\baseDatos\\temp\\Profesores.txt"));
	    	serializarMaterias(gestor,new File("src\\baseDatos\\temp\\Materias.txt"));
	    	serializarBeca(gestor,new File("src\\baseDatos\\temp\\Beca.txt"));
	    }

	    public static void serializarEstudiantes(gestionDatos gestor, File ruta) {
	        try {
	            PrintWriter pw = new PrintWriter(ruta);
	            FileOutputStream file = new FileOutputStream(ruta);
	            ObjectOutputStream out = new ObjectOutputStream(file);
	            out.writeObject(gestor.getEstudiantes());
	            out.close();
	            file.close();

	        } catch (IOException e) {
	            System.out.println("Error en la serializacion" + e);
	        }
	    }

	    public static void serializarProfesores(gestionDatos gestor, File ruta) {
	        try {
	            PrintWriter pw = new PrintWriter(ruta);
	            FileOutputStream file = new FileOutputStream(ruta);
	            ObjectOutputStream out = new ObjectOutputStream(file);
	            out.writeObject(gestor.getProfesores());
	            out.close();
	            file.close();

	        } catch (IOException e) {
	            System.out.println("Error en la serializacion" + e);
	        }
	    }

	    public static void serializarMaterias(gestionDatos gestor, File ruta) {
	        try {
	            PrintWriter pw = new PrintWriter(ruta);
	            FileOutputStream file = new FileOutputStream(ruta);
	            ObjectOutputStream out = new ObjectOutputStream(file);
	            
	            
	            out.writeObject(gestor.getMaterias());
	            out.close();
	            file.close();

	        } catch (IOException e) {
	            System.out.println("Error en la serializacion" + e);
	        }
	    }

	    public static void serializarBeca(gestionDatos gestor, File ruta) {
	        try {
	            PrintWriter pw = new PrintWriter(ruta);
	            FileOutputStream file = new FileOutputStream(ruta);
	            ObjectOutputStream out = new ObjectOutputStream(file);
	            out.writeObject(gestor.getSistemaBecas());
	            out.close();
	            file.close();

	        } catch (IOException e) {
	            System.out.println("Error en la serializacion" + e);
	        }
	    }

	}

File: src_py/baseDatos/Serializador.py
import pickle
import os

class Serializador:
    @staticmethod
    def serializar(gestor):
        base_path = os.path.dirname(os.path.abspath(__file__))
        temp_path = os.path.join(base_path, "temp")
        os.makedirs(temp_path, exist_ok=True)  # Create 'temp' directory if it doesn't exist

        est_path = os.path.join(temp_path, "Estudiantes.pkl")
        prof_path = os.path.join(temp_path, "Profesores.pkl")
        mat_path = os.path.join(temp_path, "Materias.pkl")
        beca_path = os.path.join(temp_path, "Beca.pkl")

        Serializador.serializarEstudiantes(gestor, est_path)
        Serializador.serializarProfesores(gestor, prof_path)
        Serializador.serializarMaterias(gestor, mat_path)
        Serializador.serializarBeca(gestor, beca_path)
    @staticmethod
    def serializarEstudiantes(gestor, ruta):
        try:
            with open(ruta, "wb") as file:
                pickle.dump(gestor.getEstudiantes(), file)
        except FileNotFoundError as e:
            print("Error en la serialización:", e)

    @staticmethod
    def serializarProfesores(gestor, ruta):
        try:
            with open(ruta, "wb") as file:
                pickle.dump(gestor.getProfesores(), file)
        except FileNotFoundError as e:
            print("Error en la serialización:", e)

    @staticmethod
    def serializarMaterias(gestor, ruta):
        try:
            with open(ruta, "wb") as file:
                pickle.dump(gestor.getMaterias(), file)
        except FileNotFoundError as e:
            print("Error en la serialización:", e)

    @staticmethod
    def serializarBeca(gestor, ruta):
        try:
            with open(ruta, "wb") as file:
                pickle.dump(gestor.getSistemaBecas(), file)
        except FileNotFoundError as e:
            print("Error en la serialización:", e)
